### PR TITLE
Water pollution data

### PR DESCRIPTION
**Pull request by gusteran**
- Adds a servlet to handle get requests for water pollution
- Adds a data model for the water system and the water contaminants
- Add a test to ensure data scraping from EPA API does not run into errors
- Add html elements on top of the Map to display water contaminants

Upon entering a valid zip code the user will now see:
- the water systems in their town
- how many people each serve
- the pollutants that there has been a violation of for each system

### CODE REVIEW

File: src/main/java/com/google/sps/data/WaterContaminant.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package com.google.sps.data;
```
Comment (ekengni) [inline src/main/java/com/google/sps/data/WaterContaminant.java:1]: I dont know you intended to have all the lines commented out. But if that was your intention, i think it would be best to leave it for another PR.

File: src/main/java/com/google/sps/data/WaterSystem.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    /* This class is nested static so that Gson
+    * can turn a WaterSystem into a Json 
+     */
+    public static class WaterContaminant {
+
+        private int contaminantCode;
+        private String contaminantName;
+        private String sources;
+        private String definition;
+        private String healthEffects;
```
Comment (ekengni) [inline src/main/java/com/google/sps/data/WaterSystem.java:143]: nit: comment of what violations contains.

File: src/main/java/com/google/sps/data/WaterSystem.java [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    //         return;
+    //     }
+    //     System.out.println(violationCells[7] + " in "+pwsid);
+    //     String contaminantName = violationCells[7];
+    //     contaminantsMap.putIfAbsent(contaminantName, new WaterContaminant(violationCells));
+    //     String violationDate = violationCells[18];
+    //     String enforcementAction = violationCells[17];
+    //     contaminantsMap.get(contaminantName).addViolationInstance(violationDate, enforcementAction);
+    //     System.out.println(contaminantsMap.get(contaminantName));
+    // }
```
Comment (natalie-abreu) [inline src/main/java/com/google/sps/data/WaterSystem.java:117]: As Erick noted, take out the commented lines that aren't relevant if you can

**Comment by lazau-g** [comment]
Added @lazau-g @kavinkarthik92 at @gusteran's request.

File: src/main/java/com/google/sps/servlets/WaterPollutionServlet.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+    public static final int DEFAULT_SCORE = 100;
+
+    @Override
+    public void init() {
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String area = request.getParameter(AREA_PARAMETER);
```
Comment (lazau-g) [inline src/main/java/com/google/sps/servlets/WaterPollutionServlet.java:63]: Are you planning to bring back querying by zip code/area parameter in the future? If no, remove the associated code.

File: src/main/java/com/google/sps/servlets/WaterPollutionServlet.java [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    */
+    public ArrayList<WaterSystem> retrieveSDWViolations(String town, String state){
+        ArrayList<WaterSystem> waterSystems;
+        try {
+            URL url = new URL(EPA_WATERSYSTEM_LINK+EPA_CITY_PARAMETER+town.toUpperCase()+EPA_STATE_PARAMETER+state + CSV_FORMAT);
+            // URL url = new URL(EPA_API_LINK + areaType + areaCode + MIN_DATE + CSV_FORMAT);
+            waterSystems = parseViolationsFromURL(url);
+        } catch (IOException e){
+            logger.error(e.getMessage());
+            waterSystems = new ArrayList<>();
```
Comment (lazau-g) [inline src/main/java/com/google/sps/servlets/WaterPollutionServlet.java:94]: A better way to handle this error would be to trickle it back up to the caller (WaterPollutionServlet.java line 75), catch the error there and return a 500 (Internal Server Error) response back to the caller. You can move the error logging mechanism up there too.

Returning a 500 error makes it easier to see at the browser why certain pieces of data are not loading correctly, e.g. through Chrome's dev tools. As implemented, an error here will manifest as an empty response to the original request - not so helpful for debugging.

File: src/main/webapp/map.js [hunk 6]
```diff
@@ -135,4 +142,22 @@ function iconUrl(score){
     else url +="red";
 	url += "-dot.png";
 	return url;
+}
+
+function addWaterSystem(areaType, zipCode){
+    fetch('/water?area='+areaType+"&zip_code="+zipCode+"&town="+town+"&state="+state).then(response => response.json()).then((systems) => {
```
Comment (lazau-g) [inline src/main/webapp/map.js:148]: It seems like your doGet handler in WaterPollutionServlet.java only uses the `town` and `state` parameters. Perhaps drop the other ones, or add handling for `area` and `zip_code` parameters back to doGet. This comment repeats the other comment I wrote on WaterPollutionServlet.java line 63.

**Comment by lazau-g** [comment]
Good job! Also a minor nit: remove the commented out lines that you've left sprinkled here and there to improve code cleanliness. You can always stash them away locally in a separate branch and bring them back if you want them for debugging.
